infra/main.bicep

```diff
@@ -66,7 +66,17 @@ param openAiApiOrganization string = ''
 
 param formRecognizerServiceName string = ''
 param formRecognizerResourceGroupName string = ''
-param formRecognizerResourceGroupLocation string = location
+// Limited regions for new version:
+// https://learn.microsoft.com/azure/ai-services/document-intelligence/concept-layout
+@description('Location for the Document Intelligence resource group')
+@allowed(['eastus', 'westus2', 'westeurope'])
```

Comment:
Thanks for fixing this. What is the behavior when a customer with an existing deployment redeploys? Does it prompt for their location or pick it up from their existing deployment? Do we have a workaround for customers who don't want to recreate their document intelligence?

Reply:
By default, if you re-deploy, it will prompt you to select a location and it will create a brand new Document Intelligence.

However, if a customer knows they want to reuse an existing resource that's in the right region, they can set these env vars:

```
azd env set AZURE_FORMRECOGNIZER_SERVICE existing-name-here
azd env set AZURE_FORMRECOGNIZER_LOCATION location-here
```

I don't think I can do anything more clever than that in the constraints of the Bicep language.  

I've updated the README with that and can mention in the issues and release notes. I'm also testing on my various azd environments.

Reply:
Update: I realize that this is a good time to move away from using "form recognizer" as the variable name, so I've made that change in this PR. If someone wants to use their existing service in the right region, they can follow the README steps, which now reflect the new names.
Otherwise, it deploys a new Document Intelligence and role. They can delete the remaining one or ignore it, as pricing should be 0 for a non-used resource.

Reply:
Got it - you are prompted to create a new document intelligence resource on next reprovision, unless you explicitly move.